### PR TITLE
repository: add progress bar

### PR DESCRIPTION
**Pull request by iaguis**
Use ioprogress library to show a progess bar when downloading images
from a Docker repository.

For this to work correctly we need coreos/ioprogress#3

### CODE REVIEW

File: lib/backend/repository/repository.go [hunk 1]
```diff
@@ -275,19 +266,61 @@ func getLayer(imgID, registry string, repoData *RepoData, imgSize int64) (io.Rea
...
+	if err != nil {
+		return nil, fmt.Errorf("error creating layer: %v", err)
+	}
+
+	_, err = io.Copy(layerFile, progressReader)
+	if err != nil {
+		return nil, fmt.Errorf("error getting layer: %v", err)
+	}
+
+	layerFile.Sync()
```
Comment (jonboulle) [inline lib/backend/repository/repository.go]: err?

**Comment by jonboulle** [comment]
:+1: nice! Maybe you could add example output to the readme?

**Comment by iaguis** [comment]
Now that coreos/ioprogress#3 is merged we can have fancy progress bars. PTAL.

**Comment by jonboulle** [comment]
LGTM

**Pull request merged**